internal/domain/usecases/attachment_usecase.go

```diff
@@ -123,5 +123,19 @@ func (u *attachmentUsecase) uploadAndSaveAttachments(fileReaders map[string]io.R
 }
 
 func (u *attachmentUsecase) DeleteAttachment(ID string) *apperror.AppError {
+	_, err := u.attachmentRepository.FindAttachmentByID(ID)
+	if err != nil {
+		u.logger.Named("DeleteAttachment").Error("Find attachment by ID: ", zap.Error(err))
+		return apperror.InternalServerError(fmt.Sprintf("failed to find attachment: %s", err.Error()))
+	}
```

Comment:
```go
if err != nil {
		if errors.Is(err, gorm.ErrRecordNotFound) {
			// not found error
		}
		u.logger.Named("DeleteAttachment").Error("Find attachment by ID: ", zap.Error(err))
		return apperror.InternalServerError(fmt.Sprintf("failed to find attachment: %s", err.Error()))
	}
```

imho, most errors in find are not found, so let's have a separate handling of 404 apart from 500

Reply:
agree with @bookpanda na, and you can use this below pattern instead
```go
	if _, err := u.attachmentRepository.FindAttachmentByID(ID); err != nil {
		if errors.Is(err, gorm.ErrRecordNotFound) {
			// not found error
		}
		u.logger.Named("DeleteAttachment").Error("Find attachment by ID: ", zap.Error(err))
		return apperror.InternalServerError(fmt.Sprintf("failed to find attachment: %s", err.Error()))
	}
```